pkg/controller/master/rancher/feature.go

```diff
@@ -0,0 +1,41 @@
+package rancher
+
+import (
+	"reflect"
+
+	rancherv3api "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+)
+
+var lockedFeatures = map[string]bool{
+	"harvester":                true,
+	"rke2":                     true,
+	"multi-cluster-management": true,
+}
+
+func (h *Handler) RancherFeatureOnChange(key string, feature *rancherv3api.Feature) (*rancherv3api.Feature, error) {
```

Comment:
can we add a comment to describe what this controller is doing and why it is needed, thanks.

Reply:
Updated it. Thanks.